tests/lib/c_lib/src/test_strerror.c

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2022 Meta
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <errno.h>
+#include <string.h>
+
+#include <ztest.h>
+
+ZTEST(test_c_lib, test_strerror)
+{
+	const char *expected;
+	const char *actual;
+
+	errno = 4242;
+	if (IS_ENABLED(CONFIG_MINIMAL_LIBC) &&
+	    !IS_ENABLED(CONFIG_MINIMAL_LIBC_STRING_ERROR_TABLE)) {
+		expected = "";
+		actual = strerror(EINVAL);
+
+		zassert_equal(0, strcmp("", strerror(EINVAL)), "");
+	} else {
+		expected = "Invalid argument";
+		actual = strerror(EINVAL);
+
+		zassert_equal(0, strcmp(expected, actual),
+			      "mismatch: exp: %s act: %s", expected, actual);
+	}
+
+	/* do not change errno on success */
+	zassert_equal(4242, errno, "");
+
+	/* consistent behaviour w.r.t. errno with invalid input */
+	errno = 0;
+	expected = "";
+	actual = strerror(-42);
+	zassert_equal(0, strcmp(expected, actual), "mismatch: exp: %s act: %s",
+		      expected, actual);
+	actual = strerror(4242);
+	zassert_equal(0, strcmp(expected, actual), "mismatch: exp: %s act: %s",
+		      expected, actual);
+	zassert_equal(0, errno, "");
+
+	/* consistent behaviour for "Success" */
+	if (IS_ENABLED(CONFIG_PICOLIBC) || IS_ENABLED(CONFIG_NEWLIB_LIBC) ||
```

Comment:
Could this use the same pattern as the other MINIMAL_LIBC checks and avoid explicit checks for other C libraries?

Reply:
Oof. Negative booleans are often a source of problems, but in this case, it does sound like that would make the code easier to review.